packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -127,23 +137,85 @@ Future<void> checkGradleDependencies() async {
   progress.stop();
 }
 
+/// Tries to create [settings_aar.gradle] in an app project by removing the subprojects
+/// from the existing [settings.gradle] file. This operation will fail if the existing
+/// [settings.gradle] file has local edits.
+void _createSettingsAarGradle(Directory androidDirectory) {
+  final File newSettingsFile = androidDirectory.childFile('settings_aar.gradle');
+  if (newSettingsFile.existsSync()) {
+    return;
+  }
+  final File currentSettingsFile = androidDirectory.childFile('settings.gradle');
+  if (!currentSettingsFile.existsSync()) {
+    return;
+  }
+  final String currentFileContent = currentSettingsFile.readAsStringSync();
+
+  final String newSettingsRelativeFile = fs.path.relative(newSettingsFile.path);
+  final Status status = logger.startProgress('✏️  Creating `$newSettingsRelativeFile`...',
+      timeout: timeoutConfiguration.fastOperation);
+
+  final String flutterRoot = fs.path.absolute(Cache.flutterRoot);
+  final File deprecatedFile = fs.file(fs.path.join(flutterRoot, 'packages','flutter_tools',
+      'gradle', 'deprecated_settings.gradle'));
+  assert(deprecatedFile.existsSync());
+  // Get the `settings.gradle` content variants that should be patched.
+  final List<String> deprecatedFilesContent = deprecatedFile.readAsStringSync().split(';EOF');
+  bool exactMatch = false;
+  for (String deprecatedFileContent in deprecatedFilesContent) {
+    if (currentFileContent.trim() == deprecatedFileContent.trim()) {
+      exactMatch = true;
+      break;
+    }
+  }
+  if (!exactMatch) {
+    status.cancel();
+    printError('*******************************************************************************************');
+    printError('Flutter tried to create the file `$newSettingsRelativeFile`, but failed.');
+    // Print how to manually update the file.
+    printError(fs.file(fs.path.join(flutterRoot, 'packages','flutter_tools',
+        'gradle', 'manual_migration_settings.gradle.md')).readAsStringSync());
+    printError('*******************************************************************************************');
+    throwToolExit('Please create the file and run this command again.');
+  }
+  // Copy the new file.
+  fs.file(fs.path.join(flutterRoot, 'packages','flutter_tools',
+      'gradle', 'new_settings.gradle')).copySync(newSettingsFile.path);
+  status.stop();
+  printStatus('✅ `$newSettingsRelativeFile` created successfully.');
+}
+
 // Note: Dependencies are resolved and possibly downloaded as a side-effect
 // of calculating the app properties using Gradle. This may take minutes.
-Future<GradleProject> _readGradleProject() async {
+Future<GradleProject> _readGradleProject({bool isLibrary = false}) async {
```

Comment:
According to https://codecov.io/gh/flutter/flutter/pull/35217/diff#D21-190, this has no test coverage